source/slang/slang-ir-legalize-global-values.cpp

```diff
@@ -0,0 +1,231 @@
+#include "slang-ir-legalize-global-values.h"
+
+#include "slang-ir-clone.h"
+#include "slang-ir-util.h"
+
+namespace Slang
+{
+
+void GlobalInstInliningContextGeneric::inlineGlobalValuesAndRemoveIfUnused(IRModule* module)
+{
+    Dictionary<IRInst*, List<IRUse*>> globalInstUsesToInline;
+
+    for (auto globalInst : module->getGlobalInsts())
+    {
+        if (isInlinableGlobalInst(globalInst))
+        {
+            for (auto use = globalInst->firstUse; use; use = use->nextUse)
+            {
+                if (getParentFunc(use->getUser()) != nullptr)
+                {
+                    globalInstUsesToInline.getOrAddValue(globalInst, List<IRUse*>()).add(use);
+                }
+            }
+        }
+    }
+
+    for (auto globalInstUses : globalInstUsesToInline)
+    {
+        auto inst = globalInstUses.first;
+        auto uses = globalInstUses.second;
```

Comment:
This is creating an unnecessary copy of the list.

Reply:
Thanks, changing to `auto&`. (I haven't been using `auto` much.)